packages/components/src/hooks/use-hover.ts

```diff
@@ -1,9 +1,9 @@
-import React from 'react';
+import React, { RefObject } from 'react';
 
-export const useHover = (refSetter, should_prevent_bubbling) => {
+export const useHover = (refSetter: RefObject<HTMLElement>, should_prevent_bubbling: boolean) => {
     const [value, setValue] = React.useState(false);
     const default_ref = React.useRef(null);
-    const ref = refSetter || default_ref;
+    const ref = refSetter ?? default_ref;
```

Comment:
```suggestion
    const ref = refSetter || default_ref;
```
@likhith-deriv  I would consider leaving it as it is unless we have an existing issue that has to be fixed using ?? 🙂